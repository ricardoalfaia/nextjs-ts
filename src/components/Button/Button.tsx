import React from 'react'

type Props = {
  handleClick: () => void
  label: string
}

const Button: React.FC<Props> = ({ handleClick, label }) => {
  return (
    <button data-testid="btn-1" onClick={handleClick}>
      {label}
    </button>
  )
}

export default Button
