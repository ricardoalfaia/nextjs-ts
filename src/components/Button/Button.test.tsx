import React from 'react'
import { cleanup, render, fireEvent } from '@testing-library/react'
import Button from './Button'

describe('My Button', () => {
  afterEach(cleanup)

  test('Render', async () => {
    const onClickMock = jest.fn()
    const { getByTestId, queryByText, rerender } = render(
      <Button handleClick={onClickMock} label="ok" />
    )
    const buttondNode = getByTestId('btn-1')
    fireEvent.click(buttondNode)
    expect(onClickMock).toHaveBeenCalledTimes(1)
    expect(queryByText(/ok/i)).toBeTruthy()
    rerender(<Button label="go" />)
    expect(queryByText(/go/i)).toBeTruthy()
  })

  test('FocusInput matches snapshot', () => {
    const onClickMock = jest.fn()
    const { container } = render(
      <Button handleClick={onClickMock} label="ok" />
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
