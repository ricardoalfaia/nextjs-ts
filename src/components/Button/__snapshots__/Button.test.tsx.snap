// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Button FocusInput matches snapshot 1`] = `
<button
  data-testid="btn-1"
>
  ok
</button>
`;
